File: android/app/src/main/java/co/searchrestaurant/android/app/fetch/SearchRestaurantApi.java
package co.searchrestaurant.android.app.fetch;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * Created by hassanabid on 2/27/16.
 */
public interface SearchRestaurantApi {

    @GET("api/v1/")
    Call<SearchRestaurantResponse[]> getRestaurantsList(@Query("format") String format, @Query("location") String location,
                                               @Query("rtype") String rtype);


}

File: v2/android/app/src/main/java/dev/hassanabid/searchrestaurant/service/SearchRestaurantApi.kt
package dev.hassanabid.searchrestaurant.service

import dev.hassanabid.searchrestaurant.service.response.SearchRestaurantResponse
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query


/**
 * Fetch all the nearby restaurants using Foursquare and GeoCoding Api
 */

interface SearchRestaurantApi {

    @GET("api/v1/")
    suspend fun getRestaurantsList(
        @Query("format") format: String, @Query("location") location: String,
        @Query("rtype") rtype: String
    ): List<SearchRestaurantResponse>

}